experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -940,15 +609,68 @@ function getIntention(
 	return rev === undefined ? undefined : revisionMetadata.getInfo(rev).rollbackOf ?? rev;
 }
 
-function areInverseRevisions(
-	rev1: RevisionTag | undefined,
-	rev2: RevisionTag | undefined,
-	revisionMetadata: RevisionMetadataSource,
-): boolean {
-	if (rev1 === undefined || rev2 === undefined) {
-		return false;
+// TODO: Try to share more logic with the version in rebase.ts.
+function compareCellPositions(
+	baseCellId: DetachEvent,
+	baseMark: ExistingCellMark<unknown>,
+	newMark: EmptyInputCellMark<unknown>,
+	newRevision: RevisionTag | undefined,
+	cancelledInserts: Set<RevisionTag>,
+	gapTracker: GapTracker,
+): number {
+	const newId = getCellId(newMark, newRevision);
+	if (baseCellId.revision === newId?.revision) {
+		if (isNewAttach(newMark)) {
+			// There is some change foo that is being cancelled out as part of a rebase sandwich.
+			// The marks that make up this change (and its inverse) may be broken up differently between the base
+			// changeset and the new changeset because either changeset may have been composed with other changes
+			// whose marks may now be interleaved with the marks that represent foo/its inverse.
+			// This means that the base and new marks may not be of the same length.
+			// We do however know that the all of the marks for foo will appear in the base changeset and all of the
+			// marks for the inverse of foo will appear in the new changeset, so we can be confident that whenever
+			// we encounter such pairs of marks, they do line up such that they describe changes to the same first
+			// cell. This means we can safely treat them as inverses of one another.
+			return 0;
+		}
+		return baseCellId.index - newId.index;
+	}
+
+	const offsetInNew = getOffsetAtRevision(newMark.lineage, baseCellId.revision);
+	if (offsetInNew !== undefined) {
+		const baseOffset = gapTracker.getOffset(baseCellId.revision);
+		return offsetInNew <= baseOffset ? Infinity : baseOffset - offsetInNew;
```

Comment:
Add test to verify that `baseOffset - offsetInNew` should be `-Inf`

Reply:
Wrote a test which exercises this code path, but realized this issue doesn't actually affect correctness. The return value is in the correct direction and is conservative, so at worst we'll end up doing many unnecessary splits.

Reply:
Returning `-Infinity` is actually incorrect since `newMark`'s cells can be in the middle of `baseMark`'s cells.